base/inc/array-utils.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * From this SO answer: https://stackoverflow.com/a/18781630/3710600
+ *
+ * Works with both integer and string positions.
+ *
+ * @param array      $array
+ * @param int|string $position
+ * @param mixed      $insert
+ */
+function array_insert( &$array, $position, $insert ) {
```

Comment:
We'd need to prefix this with something SiteOrigin related so we don't get any fatal error conflicts with other plugins.

Reply:
:+1: